packages/babel-parser/src/plugins/estree.ts

```diff
@@ -573,8 +573,21 @@ export default (superClass: typeof Parser) =>
     stopParseSubscript(base: N.Expression, state: N.ParseSubscriptState) {
       const node = super.stopParseSubscript(base, state);
       if (state.optionalChainMember) {
+        // Todo: refactor TSInstantiationExpression parsing such that
+        // ChainExpression is created before TSInstantiationExpression
         if (node.type === "TSInstantiationExpression") {
           node.expression = this.estreeParseChainExpression(node.expression);
+          if (process.env.BABEL_8_BREAKING) {
+            this.resetEndLocation(
```

Comment:
This change is not sufficient, we have to fix the end location of the member expression as well. At this point it seems we have to refactor TSInstantiationExpression parsing anyway.